Added authz strategy by date/time

mmoayyed opened the pull request:
Closes https://github.com/Jasig/cas/issues/1188

Allow authorization to enforce a start/end datetime optionally. Service access is only allowed if the current time is within the start/end times. Optionally, there is a separate redirect url define per service that CAS can redirect to, if access is not allowed to the service. 

Docs and UI will be updated shortly.

leleuj commented on cas-server-core/src/main/java/org/jasig/cas/services/DefaultRegisteredServiceAccessStrategy.java:
```diff
@@ -60,6 +62,12 @@
     /** Is the service allowed to use SSO? **/
     private boolean ssoEnabled = true;
 
+    private String startingDateTime;
```
Shouldn't we have a `TimeBasedRegisteredServiceAccessStrategy` for this need in addition to the `DefaultRegisteredServiceAccessStrategy`?

mmoayyed replied:
With the current design, you can only plug in one access strategy into a service.  Are you saying there could be `TimeBasedRegisteredServiceAccessStrategy` that extends from `DefaultRegisteredServiceAccessStrategy`?

leleuj replied:
Yes, that's what I meant: a new `TimeBasedRegisteredServiceAccessStrategy` extending the default one...

mmoayyed replied:
Sounds good. Will update.

leleuj commented:
+1 with a comment

mmoayyed commented:
Added `TimeBasedRegisteredServiceAccessStrategy`. Updated docs and the UI. 

![image](https://cloud.githubusercontent.com/assets/1205228/10444399/3f69c804-7123-11e5-8cbe-30f99ef35ad7.png)

The date format is customizable, the datepicker from bootstrap and moment.js is used.
